pkg/scheduler/core/generic_scheduler.go

```diff
@@ -662,46 +662,18 @@ func (g *genericScheduler) podFitsOnNode(
 	// nominated pods are running because they are not running right now and in fact,
 	// they may end up getting scheduled to a different node.
 	for i := 0; i < 2; i++ {
-		metaToUse := meta
 		stateToUse := state
 		nodeInfoToUse := info
 		if i == 0 {
 			var err error
-			podsAdded, metaToUse, stateToUse, nodeInfoToUse, err = g.addNominatedPods(ctx, pod, meta, state, info)
+			podsAdded, _, stateToUse, nodeInfoToUse, err = g.addNominatedPods(ctx, pod, meta, state, info)
```

Comment:
It looks like this is the only usage of `addNominatedPods`. Remove the return value.

Reply:
Let's get rid of it in the metadata cleanup PR.

Reply:
sg, TODO maybe?

Reply:
Yeap, added at L587:

```
// TODO(Huang-Wei): remove 'meta predicates.Metadata' from the signature.
```